pyproject.toml

```diff
@@ -85,27 +85,30 @@ make-summary-multi-line = true
 wrap-summaries = 79
 wrap-descriptions = 79
 
-[tool.flakeheaven]
-max_line_length = 88
-max_doc_length = 79
-show_source = true
+[tool.ruff]
+line-length = 88  # E501 (line-too-long)
+show-source = true
+select = [
+    "E",    # pycodestyle
+    "F",    # pyflakes
+    "I",    # isort
+    "W",    # pycodestyle warnings
+]
+ignore = ["E501"]  # Avoid enforcing line-length violations
+
+[tool.ruff.pycodestyle]
+max-doc-length = 79
 
-[tool.flakeheaven.plugins]
-pycodestyle = ["+*", "-E501", "-W503"]
-pyflakes = ["+*"]
+[tool.ruff.per-file-ignores]
+"__init__.py" = ["F401"]  # Ignore `F401` (unused-import) in all `__init__.py` files
 
-[tool.flakeheaven.exceptions."**/__init__.py"]
-pyflakes = ["-F401"]
+[tool.ruff.isort]
+known-third-party = ["pygmt"]
 
 [tool.pytest.ini_options]
 minversion = "6.0"
 addopts = "--verbose --durations=0 --durations-min=0.2 --doctest-modules --mpl --mpl-results-path=results"
 
-[tool.isort]
-profile = "black"
-skip_gitignore = true
```

Comment:
Note that `ruff`'s `isort` rules is nearly-equivalent to `isort`'s `profile="black"` settings (see https://docs.astral.sh/ruff/faq/#how-does-ruffs-import-sorting-compare-to-isort). Also, files that are gitignored are automatically skipped from being check (see https://github.com/astral-sh/ruff/pull/1234), so no need to apply these rules anymore.